core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -226,6 +226,59 @@ public void cancelOperation() throws UnsupportedOperationException {
         return this.fluxHandle.blockLast();
     }
 
+    /**
+     * Blocks indefinitely until given {@link OperationStatus} is received or a timeout expires.
+     * @param statusToBlockFor The {@link OperationStatus} to block for and it can be any valid {@link OperationStatus} except {@link OperationStatus}.
+     * @return {@link PollResponse} matching the status to block for or times out.
+     * @throws IllegalArgumentException if {@code statusToBlockFor} is {@code null}.
+     */
+    public PollResponse<T> blockUntil(PollResponse.OperationStatus statusToBlockFor) {
```

Comment:
Should have this method call into `blockUntil(OperationStatus, Duration)` with null for `Duration`. In the other method just have an if/else for `blockFirst` and `blockFirst(Duration)`, that way we only need to maintain one set of checks.